extension-testcontainer/src/main/java/io.camunda.zeebe.process.test.extension.testcontainer/ZeebeProcessTestExtension.java

```diff
@@ -165,4 +172,25 @@ private void injectField(
   private ExtensionContext.Store getStore(final ExtensionContext context) {
     return context.getStore(ExtensionContext.Namespace.create(getClass(), context.getUniqueId()));
   }
+
+  /**
+   * Get a custom object mapper from the test context
+   *
+   * @param context jUnit5 extension context
+   * @return the custom object mapper, or null if no object mapper are in the context
+   */
+  private ObjectMapper getCustomMapper(final ExtensionContext context) {
+    final Optional<Field> customMapperOpt =
+        getField(context.getRequiredTestClass(), objectMapperInstance);
+    if (!customMapperOpt.isPresent()) {
+      return null;
+    }
+    final Field customMapper = customMapperOpt.get();
+    ReflectionUtils.makeAccessible(customMapper);
+    try {
+      return (ObjectMapper) customMapper.get(context.getRequiredTestInstance());
+    } catch (final IllegalAccessException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
❌ I'm not a big fan of configuring the ObjectMapper this way. At first It thought it would be elegant, but I don't like messing with the accessibility of a field that we don't overwrite the value of (like we do with the engine and client).

I also don't believe it's sufficient to configure the ObjectMapper like this, cause it's only changed in the client now. More on that in another comment.

Reply:
Same comment goes for the other extension class of course.